src/geo/transform.js

```diff
@@ -961,6 +961,8 @@ class Transform {
 
     /**
      * Returns true if a screenspace Point p, is above the horizon.
+     * This approximates the map as an infinite plane and does not account for z0-z3
+     * wherein the map is small quad with whitespace above the north pole and below the south pole.
```

Comment:
@karimnaaji I added the explanation on why `transform.IsPointAboveHorizon` is not equivalent to `isPointOnMap` here.

This function is only used for `queryRenderedFeatures` where its used to early reject a few specific features from intersection evaluation. And this "infinite-plane" like approximation is necessary in that case because map aligned symbols and circles can extend beyond the north pole.